src/rules/importBlacklistRule.ts

```diff
@@ -54,9 +89,140 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 function walk(ctx: Lint.WalkContext<string[]>) {
+    type Options = Array<string | { [moduleName: string]: string[] }>;
+    interface BannedImports {
+        [moduleName: string]: true | Set<string>;
+    }
+
+    // Merge/normalize options.
+    // E.g., ["a", { "b": ["c"], "d": ["e"] }, "f", { "f": ["g"] }]
+    // becomes { "a": true, "b": Set(["c"]), "d": Set(["e"]), "f": true }.
+    const bannedImports = (ctx.options as Options).reduce(
+        (acc, it) => {
+            if (typeof it === "string") {
+                acc[it] = true;
+            } else {
+                Object.keys(it).forEach((moduleName) => {
+                    if (acc[moduleName] instanceof Set) {
+                        it[moduleName].forEach((bannedImport) => {
+                            (acc[moduleName] as Set<string>).add(bannedImport);
+                        });
+                    } else if (acc[moduleName] !== true) {
+                        acc[moduleName] = new Set(it[moduleName]);
+                    }
+                });
+            }
+            return acc;
+        },
+        {} as BannedImports, // tslint:disable-line no-object-literal-type-assertion
+    );
+
     for (const name of findImports(ctx.sourceFile, ImportKind.All)) {
-        if (ctx.options.indexOf(name.text) !== -1) {
-            ctx.addFailure(name.getStart(ctx.sourceFile) + 1, name.end - 1, Rule.FAILURE_STRING);
+        // TODO: Resolve/normalize relative file imports to a canonical path?
```

Comment:
how do we resolve this TODO? would prefer not to merge with such comments

Reply:
Hmm. My guess is that this isn't too hard to fix, but I didn't investigate it. The current code (i.e., pre this PR) doesn't resolve relative imports either, so fixing that seemed out of scope to adding this new feature. Therefore, I don't know if addressing that should block merge; I think I'd rather just see a new issue opened for adding that as a distinct feature.

Reply:
@ethanresnick ok please open an issue and note its number in this comment, so we know it's not just a stray TODO.

`TODO #XXX: ...`

Reply:
@giladgray Sounds good. Done!